classes/Hook.php

```diff
@@ -235,11 +236,111 @@ public static function getHookAliasList()
     }
 
     /**
+     * Get the list of hook aliases
+     *
+     * @since 1.7.1.0
+     * @return array
+     */
+    private static function getHookAliasesList()
+    {
+        $cacheId = 'hook_aliases';
+        if (!Cache::isStored($cacheId)) {
+            $hookAliasList = Db::getInstance()->executeS('SELECT * FROM `'._DB_PREFIX_.'hook_alias`');
+            $hookAliases = array();
+            if ($hookAliasList) {
+                foreach ($hookAliasList as $ha) {
+                    if (!isset($hookAliases[$ha['name']])) {
+                        $hookAliases[$ha['name']] = array();
+                    }
+                    $hookAliases[$ha['name']][] = $ha['alias'];
+                }
+            }
+            Cache::store($cacheId, $hookAliases);
+            return $hookAliases;
+        }
+        return Cache::retrieve($cacheId);
+    }
+
+    /**
+     * Return backward compatibility hook names
+     *
+     * @since 1.7.1.0
+     * @param $hookName
+     * @return array
+     */
+    private static function getHookAliasesFor($hookName)
+    {
+        $cacheId = 'hook_aliases_' . $hookName;
+        if (!Cache::isStored($cacheId)) {
+            $aliasesList = Hook::getHookAliasesList();
+
+            if (isset($aliasesList[$hookName])) {
+                Cache::store($cacheId, $aliasesList[$hookName]);
+                return $aliasesList[$hookName];
+            }
+
+            $retroName = array_keys(array_filter($aliasesList, function ($elem) use ($hookName) {
+               return in_array($hookName, $elem) ;
+            }));
+
+            if (empty($retroName)) {
+                Cache::store($cacheId, array());
+                return array();
+            }
+
+            Cache::store($cacheId, $retroName);
+            return $retroName;
+        }
+        return Cache::retrieve($cacheId);
+    }
+
+    /**
+     * Check if a hook or one of its old names is callable on a module
+     *
+     * @since 1.7.1.0
+     * @param $module
+     * @param $hookName
+     * @return bool
+     */
+    private static function isHookCallableOn($module, $hookName)
+    {
+        $aliases = Hook::getHookAliasesFor($hookName);
+        $aliases[] = $hookName;
+
+        return array_reduce($aliases, function ($prev, $curr) use ($module) {
+            return $prev || is_callable(array($module, 'hook' . $curr));
+        }, false);
+    }
+
+    /**
+     * Call a hook (or one of its old name) on a module
+     *
+     * @since 1.7.1.0
+     * @param $module
+     * @param $hookName
+     * @param $hookArgs
+     * @return string
+     */
+    private static function callHookOn($module, $hookName, $hookArgs)
+    {
+        if (is_callable(array($module, 'hook' . $hookName))) {
+            return Hook::coreCallHook($module, 'hook' . $hookName, $hookArgs);
+        }
+        foreach (Hook::getHookAliasesFor($hookName) as $hook) {
+            if (is_callable(array($module, 'hook' . $hook))) {
```

Comment:
`Hook::isHookCallableOn()` returns a boolean saying whether a hook (or one of its aliases) is callable on a module or not. Here we know that the hook is callable on the module (thanks to https://github.com/PrestaShop/PrestaShop/pull/7621/files#diff-e38418d265bff448f525a7844d93f350R828) thus we are only calling the first callable method and returning.